examples/c/common.h

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/* This should come first. */
+#define _POSIX_C_SOURCE 200809L
+
+#include <signal.h>
+
+/**
+ * @brief Signal handler.
+ *
+ * @param signum Number of received signal.
+ */
+static void sighandler(int signum)
+{
+    #ifdef __linux__
+    const char *signame = strsignal(signum);
+    fprintf(stderr, "\nReceived %s signal\n", signame);
+    #endif
+
+    #ifdef _WIN32
+    fprintf(stderr, "\nReceived %d signal\n", signum);
+    #endif
+
+    fprintf(stderr, "Exiting...\n");
+    exit(EXIT_SUCCESS);
+}
+
+/**
+ * @brief Register signal handlers.
+ */
+void reg_sighandlers()
+{
+    signal(SIGINT, sighandler);
+
+    #ifdef __linux__
+    signal(SIGQUIT, sighandler);
+    signal(SIGTSTP, sighandler);
+    #endif
+}
```

Comment:
I am in favor of sharing common functions, but instead of placing the definition in a header file, could you move this to a .c file?

Rationale: what the compiler does with definitions on header files is nonstandard. If you place this on a .c file, you are sure that you will have a single compilation unit for that function.

Reply:
Done.